Feat/centos anitya

mfocko opened the pull request:
- [x] Add an event class
- [x] Implement a parser
- [ ] Pass around versions in a list (for New Hotness we can have just one version in the list, for Anitya version update fail)
- [ ] Do not generate DB entry from the event
- [ ] **⚠️** Choosing the correct version
  - Outcomes from the arch:
    - Gather more use cases
    - **Require** the explicit setting for version mask (for CentOS at least)
- [ ] **⚠️** Decide whether to support Fedora for Anitya version update too… right now it's prefiltered for the CentOS

Related to #2068

mfocko commented on packit_service/worker/parser.py:
```diff
@@ -1568,6 +1571,37 @@ def parse_new_hotness_update_event(event) -> Optional[NewHotnessUpdateEvent]:
+            return None
+
+        # FIXME: Handle Fedora too in case we want to support multiple releases
+        # such as for Go.
```
There was an RFE for this on the Slack already…

lbarcziova replied:
+1, we can create a separate issue for this (e.g.this will imply monitoring settings for the package will not be considered anymore, therefore also changes in documentation etc.) 

we could properly test this for CentOS Stream and once it is tested to be working do the switch for Fedora packages to not break anything for already onboarded Fedora packages

mfocko commented on packit_service/worker/events/new_hotness.py:
```diff
@@ -148,3 +147,58 @@ def get_dict(self, default_dict: Optional[Dict] = None) -> dict:
+        self._versions = versions
+
+    @property
+    def version(self) -> str:
```
The major issue here is that we “generate” DB entry for the release from the event object. There's also serialization involved (for Celery).

At the same time it feels a bit weird to keep logic of “deciding the next version” in the »event«.

lbarcziova replied:
discussed together with Mato that we want all the new versions, see e.g. https://apps.fedoraproject.org/datagrepper/v2/id?id=efec10ed-9f86-4e03-9e4f-2e626934b3ae&is_raw=true&size=extra-large in here we would be interested in all 3

lbarcziova replied:
and then it is up to maintainer to define the per-branch jobs and rules (e.g. via version_udpate_mask, upstream_tag_include, upstream_tag_exclude): [example](https://src.fedoraproject.org/rpms/golang/blob/rawhide/f/.packit.yaml)

mfocko commented on packit_service/worker/events/new_hotness.py:
```diff
@@ -148,3 +147,58 @@ def get_dict(self, default_dict: Optional[Dict] = None) -> dict:
+        # we going to choose the version?
+        #   a) latest greatest
+        #   b) next unreleased?
+        # this can be also influenced by the mask…
```
Example of a problem:

* There's a `1.0.0` release in Fedora.
* We've got releases `1.0.1`, `1.1.0` and `2.0.0` from Anitya
* Which release do we sync? The next possible (`1.0.1`) or latest-greatest `2.0.0`? Syncing all of them would create a lot of noise.
* Ideal scenario here would be having the *version mask* defined, so it would be as explicit as possible.

lbarcziova reviewed (commented):
can we also have a testcase please?

mfocko commented:
recheck

mfocko commented:
@lbarcziova rebased, added test case; left notes in comments (code) and also updated the original description; this should be ready for review and merge

lbarcziova reviewed (approved):
LGTM, thanks for the refactoring of the tests!
